backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidGL32.java

```diff
@@ -0,0 +1,237 @@
+
+package com.badlogic.gdx.backends.android;
+
+import android.annotation.TargetApi;
+import android.opengl.GLES32;
+
+import com.badlogic.gdx.graphics.GL32;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+
+@TargetApi(24)
+public class AndroidGL32 extends AndroidGL31 implements GL32 {
+
+	@Override
+	public void glBlendBarrier () {
+		GLES32.glBlendBarrier();
+	}
+
+	@Override
+	public void glCopyImageSubData (int srcName, int srcTarget, int srcLevel, int srcX, int srcY, int srcZ, int dstName,
+		int dstTarget, int dstLevel, int dstX, int dstY, int dstZ, int srcWidth, int srcHeight, int srcDepth) {
+		GLES32.glCopyImageSubData(srcName, srcTarget, srcLevel, srcX, srcY, srcZ, dstName, dstTarget, dstLevel, dstX, dstY, dstZ,
+			srcWidth, srcHeight, srcDepth);
+	}
+
+	@Override
+	public void glDebugMessageControl (int source, int type, int severity, IntBuffer ids, boolean enabled) {
+		GLES32.glDebugMessageControl(source, type, severity, ids.remaining(), ids, enabled);
+	}
+
+	@Override
+	public void glDebugMessageInsert (int source, int type, int id, int severity, String buf) {
+		GLES32.glDebugMessageInsert(source, type, id, severity, buf.length(), buf);
+	}
+
+	@Override
+	public void glDebugMessageCallback (DebugProc callback) {
+		GLES32.glDebugMessageCallback(new GLES32.DebugProc() {
+			@Override
+			public void onMessage (int source, int type, int id, int severity, String message) {
+				callback.onMessage(source, type, id, severity, message);
```

Comment:
`DebugProc callback` should be `final DebugProc callback` because libGDX continues to live in the Java 7 caves and refuses to move into the light of language level 8. But that's another issue, sort-of. This can be fixed here just by adding `final`.

Reply:
`DebugProc callback` is now final!